src/storage.rs

```diff
@@ -24,53 +27,282 @@ pub async fn init() -> sqlx::MySqlPool {
 
 #[async_trait]
 pub trait MediatorPersistence: Send + Sync + 'static {
-    async fn persist_forward_message(&self, forward_msg: &ForwardMsg);
+    async fn create_account(&self, auth_pubkey: &str) -> Result<(), String>;
+    // async fn vaporize_account(&self, auth_pubkey: String);
+    async fn add_recipient(&self, auth_pubkey: &str, recipient_key: &str) ->  Result<(), String>;
+    async fn remove_recipient(&self, auth_pubkey: &str, recipient_key: &str) ->  Result<(), String>;
+    async fn list_recipient_keys(&self, auth_pubkey: &str) -> Result<Vec<String>, String>;
+    async fn persist_forward_message(&self, forward_msg: &ForwardMsg) -> Result<(), String>;
     async fn retrieve_pending_message_count(&self, recipient_key: Option<&String>) -> u32;
     // async fn retrieve_pending_messages(
     //     &self,
     //     limit: u32,
     //     recipient_key: Option<&String>,
-    // ) -> Vec<ForwardMsg>;
-    // async fn mark_messages_recieved(&self, message_id: Vec<u32>);
+    // ) -> Vec<(u32, Vec<u8>)>;
+    // async fn mark_messages_received(&self, message_id: Vec<u32>);
 }
 
 #[cfg(feature = "mysql_db")]
 #[async_trait]
 impl MediatorPersistence for sqlx::MySqlPool {
-    async fn persist_forward_message(&self, forward_msg: &ForwardMsg) {
-        sqlx::query("INSERT INTO forward_raw VALUES (DEFAULT, ?, ?, DEFAULT)")
+    async fn create_account(&self, auth_pubkey: &str) -> Result<(), String> {
+        info!("Adding new account to database with auth_pubkey {:#?}", &auth_pubkey);
+        let insert_res = sqlx::query("INSERT INTO accounts (auth_pubkey) VALUES (?);")
```

Comment:
This was done for 2 reasons:
- `auth_pubkey` being a "string", was not suitable to use as primary key or index. (MySQL complains)
  - The full size that a key can be isn't *explicitly* defined, so I've used STRING to store it, rather than fixed sized VARCHAR.
- To have the possibility to later change the auth_pubkey without having to create an entirely new 'account'. Basically to support let's say key rotation. 
- Another later consideration is to have a primary identifier independent of user provided personal material. For ease in satisfying GDPR etc constraints. By having an independent identifier from the start we can support full deletion (including auth_key), while still keeping some relevant data, like say total forward requests served.

Reply:
The `auth_pubkey_sha` can be useful to detect and deny duplicate registration for same `auth_key`. That's the only reason I added it. (Because iirc, STRING `auth_key` did not support UNIQUE constraint)

If switching to fixed size VARCHAR, I think this would not be needed.